register.go

```diff
@@ -84,6 +93,17 @@ func (s *Snitch) register(looper director.Looper) error {
 		// Blocks until something is received
 		cmd, err := stream.Recv()
 		if err != nil {
+			// This is the first registration attempt and it has failed.
+			// Depending on IgnoreStartupError, we may need to stop the loop
+			// and tell the caller that we failed to complete registration.
+			if initialRegister && !s.config.IgnoreStartupError {
```

Comment:
Without this, `New()` would only fail if `snitch-server` was unavailable. Auth, bad requests, etc. would not cause an err - need to perform a `Recv()` to get any other errors returned by the gRPC server.